### ISSUE TITLE
HW03 Two column layout in R

### ISSUE THREAD

**Issue by angelabkim** [STAT545-UBC/Discussion#347]
@jennybc 

I followed [this gist ](https://gist.github.com/jennybc/e9e9aba6ba18c72cec26#file-2015-03-02_plot-next-to-table-rmd) mentioned in the hw outline to create a side-by-side table & plot layout.

It works, but now my document is an html_document! Is there a way to do this directly in R and keep it as a github_document? On Stackoverflow, the solutions seem to apply only to R Presentations. 

Thank you!
Angela

**Comment by ksedivyhaley** [comment]
Also interested in this question. I tried it myself and it looked great when previewed in R Studio, but github doesn't seem to interpret the layout properly from the .md file (which I keep by setting):

```
output: 
  html_document:
    css: two_columns.css
    keep_md: true
```

Specifically, github displays blocks like this:

> ```r gapminder %>% ggplot(aes(x=continent, y=gdpPercap)) + geom_boxplot() ``` ![](hw03_ExpandedGapminderMamipulation_files/figure-html/GDP Spread graph-1.png) ```r gapminder %>% group_by(continent) %>% summarise(gdp_pc_min = min(gdpPercap), gdp_pc_mean = mean(gdpPercap), gdp_pc_max = max(gdpPercap)) %>% pander(style = "rmarkdown") ``` | continent | gdp_pc_min | gdp_pc_mean | gdp_pc_max | |:-----------:|:------------:|:-------------:|:------------:| | Africa | 241.2 | 2194 | 21951 | | Americas | 1201.6 | 7136 | 42952 | | Asia | 331.0 | 7902 | 113523 | | Europe | 973.5 | 14469 | 49357 | | Oceania | 10039.6 | 18622 | 34435 |

**Comment by angelabkim** [comment]
@ksedivyhaley @jennybc 

Hi Kate, 

You're right! I just pushed to github and I have the same problem. The two columns with kable+ggplot looked great in the knit preview, but looks exactly like yours on github, broken links and all. 

As an aside, I also tried to follow the most voted solution [here](http://stackoverflow.com/questions/31753897/2-column-section-in-r-markdown), to include the css directly into markdown, but I couldn't get it to work for me.

**Comment by jennybc** [comment]
Just skimming and reacting from instinct ... I would be surprised if there's an easy way to do this in markdown. I imagine it is one of those things that is only achievable when your target output format is HTML 😢.

**Comment by ksedivyhaley** [comment]
Bummer, it looked so pretty...

**Comment by oganm** [comment]
There is a workaround if you are ok with mingling with the purity of the markdown.

**Comment by oganm** [comment]
There is a workaround if you don't mind mingling with the purity of markdown. https://gist.github.com/oganm/abef941dd42934b5ec4a35adc6b86850.

I changed the output method of kable to html so it will be rendered correctly even if it is inside a table, and made sure I know the file name of the output plot by naming the code chunk and calling the figure with html code. The `<!--`   `-->` tags around the code chunk makes sure its regular output is not displayed which is in markdown format

**Comment by oganm** [comment]
A student just pointed me to several bugs in the code above. They are now fixed. Just remember to write the correct file address

**Comment by jennybc** [comment]
Nice gist @oganm. Now that I think about it, it's pretty similar to the hack I use to put the three Lord of the Rings tables together [here](https://github.com/jennybc/lotr-tidy/blob/master/01-intro.md). But you've got a nice example of a table in one cell and a figure in the other. I think this is a decent solution for markdown meant for display on GitHub.

**Comment by angelabkim** [comment]
That's awesome, thank you for the solution @oganm!

**Comment by ksedivyhaley** [comment]
Thanks, I didn't have time to use this fix in the last homework but I'd love to use it on the join part of this week's homework!

**Comment by angelabkim** [comment]
@oganm In your solution, the default kable table ends up getting bolded throughout (not just in the column headers as is normal). Is there a way to un-bold the contents of the table? I've tried looking up html formatting options, but I couldn't find a way that removes formatting.

**Comment by oganm** [comment]
Yes that is a problem I didn't have time to solve. I'll try to take a look later but for now you are on your own. I suspect it happens because by default the first row is bold in a table so adding a phantom first row might help but no guarantees.

**Comment by angelabkim** [comment]
I ended up using `xtable` and copied Jenny's raw code from her [lotr document](https://github.com/jennybc/lotr-tidy/blob/master/01-intro.Rmd) to generate side by side tables. It worked great! No formatting issues.
